js/flatpickr_buttons_plugin.js

```diff
@@ -0,0 +1,117 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+window.CustomFlatpickrButtons = (config = {}) => {
+
+   if (config.buttons !== undefined && !Array.isArray(config.buttons)) {
+      config.buttons = [config.buttons];
+   }
+
+   return (fp) => {
+      let wrapper;
+
+      if (config.buttons === undefined) {
+         config.buttons = [{
+            label: fp.config.enableTime ? __('Now') : __("Today"),
+            attributes: {
+               'class': 'btn btn-outline-secondary'
+            },
+            onClick: (e, fp) => {
+               fp.setDate(new Date());
+            }
+         }];
+      }
+
+      return {
+         onReady: () => {
+            wrapper = `<div class="flatpickr-custom-buttons pb-1"><div class="buttons-container">`;
+
+            (config.buttons).forEach((b, index) => {
+               const button = document.createElement('button');
+               button.type = 'button';
+               button.classList.add('ms-2');
+               button.textContent = b.label;
+               button.setAttribute('btn-id', index);
+               if (typeof b.attributes !== 'undefined') {
+                  Object.keys(b.attributes).forEach((key) => {
+                     if (key === 'class') {
+                        button.classList.add(...b.attributes[key].split(' '));
+                        return;
+                     }
+
+                     button.setAttribute(key, b.attributes[key]);
+                  });
+               }
+
+               wrapper += button.outerHTML;
+
+               fp.pluginElements.push(button);
+            });
+            wrapper += '</div></div>';
+
+            fp.calendarContainer.appendChild($.parseHTML(wrapper)[0]);
+
+            $(fp.calendarContainer).on('click', '.flatpickr-custom-buttons button', (e) => {
+               e.stopPropagation();
+               e.preventDefault();
+
+               const btn = $(e.target);
+               const btn_id = btn.attr('btn-id');
+               const click_handler = config.buttons[btn_id].onClick;
+
+               if (typeof click_handler !== 'function') {
+                  return;
+               }
+
+               click_handler(e, fp);
+            });
+
+            $(fp.calendarContainer).on('keydown', '.flatpickr-custom-buttons', (e) => {
+               const target = e.target;
+               if (target.tagName.toLowerCase() !== 'button') {
+                  return;
+               }
+
+               if (e.key === 'Tab') {
+                  if ((e.shiftKey && !target.previousSibling) || (!e.shiftKey && !target.nextSibling)) {
+                     e.preventDefault();
+                     fp.element.focus();
+                  }
+               }
+            });
```

Comment:
What is the purpose of this piece of code ? I tried to use `tab` key with and without this code, and in any case I do not achieve to focus custom buttons.

I also noticed that I was not able to focus on the Flatpickr popover using only keyboard. Indeed, if I use the tab key to go from a previous input to a datetime input, the popover is opened, but using tab again will just put the focus to the next input. I am not sure how this should be handled.
1. If we focus on the flatpickr popover, people would have to jump through all flatpicker fields to be able to access the next input.
2. If we do not focus on it, people would have to always type the datetime manually.
Enhancing the keyboard navigation inside this popover seems not really usefull, unless we found a solution to be able to focus on it using the keyboard.